### PR TITLE
Move over remaining AD tests; clean up cypress run scripts

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description

This PR moves over the remaining AD cypress tests, including detector list page tests and create detector workflow tests. Refactors the dashboard tests by adding the test suite directly in anomaly-detection folder, instead of abstracting the test in `cypress/integration/common`.

It also adds some useful scripts in `package.json` to reduce some code duplication used in the github workflows. 

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ohltyler** [comment]
@amitgalitz @ylwu-amzn feel free to add any thoughts you have here as well. A follow-up to this will be removing cypress dependency + tests from AD Dashboards plugin.

**Comment by tianleh** [comment]
Noticed the test failure which is tracked here https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/90

@Flyingliuhub is working on a PR to skip it. Once it is done, @ohltyler can merge that one to get rid of the check failure.

File: cypress/utils/constants.js [hunk 1]
```diff
@@ -18,10 +18,36 @@ export const BASE_AD_PATH = BASE_PATH + '/app/anomaly-detection-dashboards#';
...
+
+export function getADStartDetectorNodeApiPath(detectorId) {
+  return BASE_AD_NODE_API_PATH + '/detectors/' + detectorId + '/start';
+}
+
+export function getADStopDetectorNodeApiPath(detectorId) {
+  return BASE_AD_NODE_API_PATH + '/detectors/' + detectorId + '/stop';
+}
+
+export const TEST_DETECTOR_ID = 'ulgqpXEBqtadYz9j2MHG';
```
Comment (Tengda-He) [inline cypress/utils/constants.js:44]: is this value random ?

File: package.json [hunk 2]
```diff
@@ -5,7 +5,11 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "cypress:open": "cypress open"
+    "cypress:open": "cypress open",
+    "cypress:run": "env TZ=America/Los_Angeles cypress run --browser chrome --headless --env SECURITY_ENABLED=false",
```
Comment (Tengda-He) [inline package.json:9]: should this be named run-without-security

**Review by Tengda-He — APPROVED** [review]

File: cypress/utils/constants.js (hunk 1 above)
Comment (ohltyler) [reply cypress/utils/constants.js:44]: No, it's a common value used in many of the fixture files which are referenced in the tests.

But, just realized there are some missing test fixtures that was missed when transferring everything over to this repo. But the tests are still passing. Need to investigate that.

File: package.json (hunk 2 above)
Comment (ohltyler) [reply package.json:9]: I'm ok to change to that to make it more verbose and clear, sure

File: cypress/utils/constants.js [hunk 3]
```diff
@@ -18,10 +18,36 @@ export const BASE_AD_PATH = BASE_PATH + '/app/anomaly-detection-dashboards#';
...
+
+// TODO: when repo is onboarded to typescript we can convert these detector states into an enum
```
Comment (tianleh) [inline cypress/utils/constants.js:46]: Feel free to create a feature request issue.

**Review by tianleh — APPROVED** [review]

**Comment by ohltyler** [comment]
@tianleh @Tengda-He please have another quick look at the latest commit when you get the chance. Basically the `baseUrl` had a bug which was causing the AD tests to visit incorrect paths, and silently pass the tests without running them properly. When fixed, I found the issues with the fixtures, such as incorrect paths and missing files altogether. They've all been fixed now and are passing. I confirmed by running in browser locally, that the fixture data is coming up for each test.

File: cypress/utils/constants.js (hunk 1 above)
Comment (ohltyler) [reply cypress/utils/constants.js:43]: Responded in new comment: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/89#issuecomment-1034360265

File: cypress/utils/constants.js (hunk 3 above)
Comment (ohltyler) [reply cypress/utils/constants.js:45]: https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/93

**Review by tianleh — APPROVED** [review]

**Comment by tianleh** [comment]
> @tianleh @Tengda-He please have another quick look at the latest commit when you get the chance. Basically the `baseUrl` had a bug which was causing the AD tests to visit incorrect paths, and silently pass the tests without running them properly. When fixed, I found the issues with the fixtures, such as incorrect paths and missing files altogether. They've all been fixed now and are passing. I confirmed by running in browser locally, that the fixture data is coming up for each test.

Great dive deep!

**Pull request merged**